sky/resources.py

```diff
@@ -803,12 +803,16 @@ def __setstate__(self, state):
             self._zone = None
 
         if version < 6:
-            accelerators = state.pop('_accelerators')
+            accelerators = state.pop('_accelerators', None)
             if accelerators is not None:
                 accelerators = {
                     accelerator_registry.canonicalize_accelerator_name(acc):
                     acc_count for acc, acc_count in accelerators.items()
                 }
             state['_accelerators'] = accelerators
 
+        image_id = state.get('_image_id', None)
+        if isinstance(image_id, str):
+            state['_image_id'] = {state.get('_region', None): image_id}
```

Comment:
Is it right to say L814-816 is the fix, and the default value changes in lines above don't directly fix this issue?

It may be worth documenting the line [writing `None in self.image_id`](https://github.com/skypilot-org/skypilot/blob/ad6296b71f8c6dde35ed47083e920606fb4e1b00/sky/resources.py#L140) - it's a bit weird to see None being used as a dict key, and we should document the semantics?

Reply:
Yes, the L814-L816 is the fix. The default value above is to avoid non-exist states causing any error.

Good point! Added the comment in the `__init__` function. Thanks!